### PR TITLE
test: Prevent limits from being set on pods and enforce requests

### PR DESCRIPTION
**Pull request by smarterclayton**
Enforce the rules we define in CONVENTIONS.md

### CODE REVIEW

**Comment by smarterclayton** [comment]
Tagging to close the barn door

File: test/extended/operators/resources.go [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+		}
+
+		// pods that have a bug opened, every entry here must have a bug associated
+		knownBrokenPods := map[string]string{
+			//"<apiVersion>/<kind>/<namespace>/<name>/(initContainer|container)/<container_name>/<violation_type>": "<url to bug>",
+
+			"apps/v1/DaemonSet/openshift-ovn-kubernetes/ovnkube-master/container/sbdb/request[cpu]":    "https://bugzilla.redhat.com/show_bug.cgi?id=1939752",
+			"apps/v1/DaemonSet/openshift-ovn-kubernetes/ovnkube-master/container/sbdb/request[memory]": "https://bugzilla.redhat.com/show_bug.cgi?id=1939752",
+
+			"apps/v1/DaemonSet/openshift-multus/multus-admission-controller/container/multus-admission-controller/request[memory]": "https://bugzilla.redhat.com/show_bug.cgi?id=1938469",
```
Comment (wking) [inline test/extended/operators/resources.go:43]: Closed as a dup of the POST https://bugzilla.redhat.com/show_bug.cgi?id=1905331

**Pull request merged**